src/commcare_cloud/ansible/roles/elasticsearch/templates/config/elasticsearch.yml.j2

```
@@ -31,20 +31,25 @@ http.max_content_length: 200mb
 path.data: {{ elasticsearch_data_dir }}/data
 path.logs: {{ elasticsearch_data_dir }}/logs
 
-{% if elasticsearch_version == '2.4.6'%}
+{% if elasticsearch_version is version('5.6.16', '<=') %}
```

Would it make more sense to say `version('5.0.0', '>=')` here? As is, if we ever rolled out some older release of v5 for some reason, it would do the 2.4 thing, which would most likely be wrong.
```suggestion
{% if elasticsearch_version is version('5.0.0', '<') %}
```
The same applies to other version checks below.

---

5.6.16 is the last version that is released in 5 version series so 5.6.16 would cover every 5 version. And these changes are compatible with ES version < 5.6.16 including version 2. So I don't think that even if we roll out some older 5 version we would have issues. It should work fine as it is.

I have updated the template file in way that it can produce correct working config for both ES 5 and ES2 based on the variable `elasticsearch_variable`

---

Oh, I was reading the condition incorrectly. But now I wonder if it should say something like `if elasticsearch_version is version('6', '<')`? Will we want this `if` block to execute if we install some newer point release, for example a security patch?

---

That would make sense but given that ES 5 is no longer maintained so chanced of them releasing a new patch version are minimal. So I am not sure if changing that would benefit us in any way.